src/SourceBuild/content/eng/Publishing.props

```diff
@@ -0,0 +1,104 @@
+    <SourceBuiltPdbArtifactsDir>$(ArtifactsStagingDir)/PDB/$(Configuration)</SourceBuiltPdbArtifactsDir>
+    <PublishingUseHardlinksIfPossible>true</PublishingUseHardlinksIfPossible>
+    <AssetManifestFileName Condition="'$(MergedAssetManifestName)' != ''">$(MergedAssetManifestName)</AssetManifestFileName>
+  </PropertyGroup>
+  <PropertyGroup Condition="'$(DotNetBuildSourceOnly)' != 'true'">
+    <SourceBuiltShippingPackagesDir>$(ArtifactsStagingDir)/packages/$(Configuration)/Shipping</SourceBuiltShippingPackagesDir>
+    <SourceBuiltNonShippingPackagesDir>$(ArtifactsStagingDir)/packages/$(Configuration)/NonShipping</SourceBuiltNonShippingPackagesDir>
+  </PropertyGroup>
+  <!--
+    When building from source, the Private.SourceBuilt.Artifacts archive will contain the nuget packages, so don't push them where we'll upload from.
+  -->
+  <PropertyGroup Condition="'$(DotNetBuildSourceOnly)' == 'true'">
+    <SourceBuiltShippingPackagesDir>$(ArtifactsTmpDir)/staging/packages/$(Configuration)/Shipping</SourceBuiltShippingPackagesDir>
+    <SourceBuiltNonShippingPackagesDir>$(ArtifactsTmpDir)/staging/packages/$(Configuration)/NonShipping</SourceBuiltNonShippingPackagesDir>
+  </PropertyGroup>
+
+  <!--
+    Don't binplace any artifacts with vertical visibility. These are only used for building other repos within this vertical
+    and shouldn't be pushed to build storage.
+  -->
+  <ItemGroup>
+    <ArtifactVisibilityToPublish Include="Internal;External" />
+  </ItemGroup>
+
+  <!-- Build task assembly paths -->
+  <PropertyGroup>
+    <MicrosoftDotNetUnifiedBuildTasksAssembly>$([MSBuild]::NormalizePath('$(ArtifactsBinDir)', 'Microsoft.DotNet.UnifiedBuild.Tasks', '$(Configuration)', 'Microsoft.DotNet.UnifiedBuild.Tasks.dll'))</MicrosoftDotNetUnifiedBuildTasksAssembly>
+  </PropertyGroup>
+
+  <UsingTask TaskName="Microsoft.DotNet.UnifiedBuild.Tasks.GetKnownArtifactsFromAssetManifests" AssemblyFile="$(MicrosoftDotNetUnifiedBuildTasksAssembly)" TaskFactory="TaskHostFactory" />
+
+  <Target Name="DiscoverArtifacts">
+    <ItemGroup>
+      <!-- Repo manifests from individual repos -->
+      <RepoManifests Include="$(AssetManifestsIntermediateDir)/**/*.xml" />
+    </ItemGroup>
+    <!-- Get produced packages and assets from the manifests -->
+    <GetKnownArtifactsFromAssetManifests AssetManifests="@(RepoManifests)">
+      <Output TaskParameter="KnownPackages" ItemName="ProducedPackage" />
+      <Output TaskParameter="KnownBlobs" ItemName="ProducedAsset" />
+    </GetKnownArtifactsFromAssetManifests>
+    <ItemGroup>
+      <ProducedPackage>
+        <IsShipping Condition="'%(ProducedPackage.NonShipping)' != 'true'">true</IsShipping>
+        <IsShipping Condition="'%(ProducedPackage.NonShipping)' == 'true'">false</IsShipping>
+        <Kind>Package</Kind>
+        <PublishFlatContainer>false</PublishFlatContainer>
+        <PackageId>%(Identity)</PackageId>
```

Comment:
I don't recall this being a bit of metadata read off of the Artifacts. Where is this used?

Reply:
This is used in the source-build targets in `PublishSourceBuild.targets`.